src/web_parser.py

```diff
@@ -0,0 +1,73 @@
+# Log parser to create json from a log file
+
+import itertools
+import re
+import json
+
+file_path = "../test_logs/sixrobotlog/LOG_Dolphin0_1_12_2020_____13_01_34/LOG_Dolphin0_1_12_2020_____13_01_34.alog"
+opened_file = open(file_path, "r")
+
+
+# Parameter is an opened file, rather than a file path
+def web_parser(file):
+    # Check to see what type of log file this is, and set log_type and robot_id appropriately
+    if "LOG_Narwhal" in file.name:
+        print("This is a Narwhal log")
+        log_type = "Narwhal"
+        robot_id = "Narwhal"
+    elif "LOG_Dolphin" in file_path:
+        print("This is a Dolphin log")
+        log_type = "Dolphin"
+        # Extract dolphin id
+        robot_match = re.search(r'Dolphin\d+', file.name)
+        if robot_match.group(0):
+            robot_id = robot_match.group(0)
+        else:
+            print("Error searching for robot id")
+    else:
+        print('Log neither Dolphin nor Narwhal')
+
+    # Parse date and time from file path
+    # The leading _ differentiates the date and time from robot id
+    matches = re.findall(r'_[0-9]+_[0-9]+_[0-9]+', file.name)
+
+    # Split the date and time from the underscores
+    date_parts = matches[0].split('_')
+    time_parts = matches[1].split('_')
+
+    # Recreate the date and times with appropriate separator
+    date = '-'.join(date_parts[1:])
+    time = ':'.join(time_parts[1:])
+
+    # Parsed script data
+    parsed = {
+        "robot_id": robot_id,
```

Comment:
It would probably be better for `robot_id` to be called something else here since `Narwhal` could also be the value here

Reply:
This also applies to the `robot_id` variable(s) above when attempting to find if the log is a Dolphin or Narwhal

Reply:
My thinking was the Narwhal is still a robot, just not a Dolphin.

Reply:
The Narwhal is the ground control station that all of the robots communicate with, not really a robot itself